crates/starfish/core/src/universal_committer.rs

```diff
@@ -181,29 +183,14 @@ pub(crate) mod universal_committer_builder {
                 dag_state,
                 wave_length: DEFAULT_WAVE_LENGTH,
                 number_of_leaders: 1,
-                pipeline: false,
             }
         }
 
-        #[expect(unused)]
-        pub(crate) fn with_wave_length(mut self, wave_length: Round) -> Self {
-            self.wave_length = wave_length;
-            self
-        }
-
-        pub(crate) fn with_number_of_leaders(mut self, number_of_leaders: usize) -> Self {
```

Comment:
I'd leave this one, as we want to keep the possibility of changing the number of leaders for the future.